drivers/spi/spi_nrfx_common.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2023, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "spi_nrfx_common.h"
+#include <zephyr/kernel.h>
+#include <nrfx_gpiote.h>
+
+int spi_nrfx_wake_init(uint32_t wake_pin)
+{
+	nrfx_gpiote_input_config_t input_config = {
+		.pull = NRF_GPIO_PIN_PULLDOWN,
+	};
+	uint8_t ch;
+	nrfx_gpiote_trigger_config_t trigger_config = {
+		.trigger = NRFX_GPIOTE_TRIGGER_HITOLO,
+		.p_in_channel = &ch,
+	};
+	nrfx_err_t res;
+
+	res = nrfx_gpiote_channel_alloc(&ch);
+	if (res != NRFX_SUCCESS) {
+		return -ENODEV;
+	}
+
+	res = nrfx_gpiote_input_configure(wake_pin,
```

Comment:
i wonder if it would be good to put somewhere the information that if it fails it's most probably due to `CONFIG_NRFX_GPIOTE_NUM_OF_EVT_HANDLERS` being to low. It might be difficult for the user to figure that out.

Reply:
Handler is not used here, only the allocated GPIOTE channel as an event generator (the event is checked in a loop).